cms/cache/page.py

```diff
@@ -45,30 +45,40 @@ def set_page_cache(response):
 
     placeholders = toolbar.content_renderer.get_rendered_placeholders()
     # Checks if there's a plugin using the legacy "cache = False"
-    placeholder_ttl_list = []
+    ttl_list = []
     vary_cache_on_set = set()
     for ph in placeholders:
         # get_cache_expiration() always returns:
         #     EXPIRE_NOW <= int <= MAX_EXPIRATION_IN_SECONDS
         ttl = ph.get_cache_expiration(request, timestamp)
         vary_cache_on = ph.get_vary_cache_on(request)
 
-        placeholder_ttl_list.append(ttl)
+        ttl_list.append(ttl)
         if ttl and vary_cache_on:
             # We're only interested in vary headers if they come from
             # a cache-able placeholder.
             vary_cache_on_set |= set(vary_cache_on)
 
-    if EXPIRE_NOW not in placeholder_ttl_list:
-        if placeholder_ttl_list:
-            min_placeholder_ttl = min(x for x in placeholder_ttl_list)
-        else:
-            # Should only happen when there are no placeholders at all
-            min_placeholder_ttl = MAX_EXPIRATION_TTL
-        ttl = min(
-            get_cms_setting('CACHE_DURATIONS')['content'],
-            min_placeholder_ttl
-        )
+    if EXPIRE_NOW not in ttl_list:
+        ttl_list.append(get_cms_setting('CACHE_DURATIONS')['content'])
+        ttl_list.append(MAX_EXPIRATION_TTL)
+
+        # If defined the `CMS_CACHE_LIMIT_TTL_CLASS` extension point
+        if hasattr(settings, 'CMS_CACHE_LIMIT_TTL_CLASS'):
+            # then loads the class
+            class_name = settings.CMS_CACHE_LIMIT_TTL_CLASS
+
+            import importlib
+            module_name, class_name = class_name.rsplit(".", 1)
+            LimitPageCacheTTL = getattr(importlib.import_module(module_name), class_name)
+            instance = LimitPageCacheTTL()
```

Comment:
Why not directly point the setting to a function?

Reply:
Yeah this might be a bit neater as a utility function and it might avoid the local import that way.

Reply:
@sampaccoud  and @marksweb  I've reviewed the PR. Change the extension point to reference a function instead of a class. I've also renamed the extension point has: `CMS_LIMIT_TTL_CACHE_FUNCTION`